### PR TITLE
Add `inference_mode` hint message to use `eval` with inference.

### LINKED ISSUE #162923: Method `.inference_mode()` can be misleading
### 📚 The doc issue

Hi

The name of [inference_mode](https://docs.pytorch.org/docs/stable/generated/torch.autograd.grad_mode.inference_mode.html) can be confusing.
At first glance, the name suggests that it might also handle tasks commonly associated with inference, such as disabling dropout or switching batch normalization layers to evaluation mode. While the documentation does not mention such behavior, this could easily lead to confusion for users (as it did for me).

### Suggest a potential alternative/fix

The easiest solution would be to add a clear note in the documentation stating that, for actual inference, the model should also be set to evaluation mode using `.eval()`. This would help prevent users from assuming that `inference_mode` handles those responsibilities automatically.

cc @svekars @sekyondaMeta @AlannaBurke @albanD

### PR DESCRIPTION
**Pull request by zeshengzong**
Fixes #162923

### Before

image

### After

image

### CODE REVIEW

**Review by jbschlosser — APPROVED** [review]
Looks great, thanks for the update!
